src/handlers.rs

```diff
@@ -9,13 +9,8 @@ use futures::future::{self, Future};
 use serde::de::{Deserialize, Deserializer};
 
 use auth::HawkPayload;
-use db::{params, util::ms_since_epoch, Db, DbError};
-
-/// This is the global HTTP state object that will be made available to all
-/// HTTP API calls.
-pub struct ServerState {
-    pub db: Box<Db>,
-}
+use db::{params, util::ms_since_epoch, DbError};
+use server::ServerState;
```

Comment:
Now that `ServerState` is referenced in a few different places, it seemed to belong more in the `server` module than in `handlers`.